src/sentry/integrations/slack/notify_action.py

```diff
@@ -112,17 +115,14 @@ def send_notification(event, futures):
                 "attachments": json.dumps([attachment]),
             }
 
-            session = http.build_session()
-            resp = session.post("https://slack.com/api/chat.postMessage", data=payload, timeout=5)
-            status_code = resp.status_code
-            response = resp.json()
-            track_response_code(status_code, response.get("ok"))
-            resp.raise_for_status()
-            if not response.get("ok"):
+            client = SlackClient()
+            try:
+                client.post("/chat.postMessage", data=payload, timeout=5)
+            except ApiError as e:
                 self.logger.info(
                     "rule.fail.slack_post",
                     extra={
-                        "error": response.get("error"),
+                        "error": six.text_type(e),
                         "project_id": event.project_id,
                         "event_id": event.event_id,
```

Comment:
@scefali still going to add the channel in here

Reply:
@MeredithAnya Well, we want to add the channel on `SlackClient` as `logging_context`. This log message is going to be somewhat redundant since we'll already log the error at the client level.

Reply:
looking at it more, adding the channel to the logging_context doesn't seem as helpful to me 

 ```
self.logger.info(u"%s.http_response" % (self.integration_type), extra=extra)
```

If an alert rule for slack is failing, I rather look up the logs using `rule.fail.slack_post` instead of `slack.http_response` and having to figure out which logs have to do with the alert rule.

Reply:
@MeredithAnya could we do it for both? Sometimes we might look at the `http_response` I think.